packages/flutter/test/widgets/safe_area_test.dart

```diff
@@ -89,6 +89,21 @@ void main() {
       expect(tester.getBottomRight(find.byType(Placeholder)), const Offset(800.0, 600.0));
     });
 
+    testWidgets('SafeArea - properties', (WidgetTester tester) async {
+      final SafeArea child = SafeArea(
+        left: true,
+        right: false,
+        bottom: false,
+        child: Container()
+      );
+      final DiagnosticPropertiesBuilder properties = DiagnosticPropertiesBuilder();
+      child.debugFillProperties(properties);
+
+      expect(properties.properties.any((DiagnosticsNode n) => n is FlagProperty && n.toString() == 'avoid left padding'), true);
+      expect(properties.properties.any((DiagnosticsNode n) => n is FlagProperty && n.toString() == 'avoid top padding'), true);
+      expect(properties.properties.any((DiagnosticsNode n) => n is FlagProperty && n.toString() == 'avoid bottom padding'), false);
```

Comment:
Covers basis of specifying true or false options, or using default (null) option

Reply:
Let's check the right padding as well since it is part of the change.

Reply:
same for sliver one